python/ribasim/ribasim/model.py

```diff
@@ -80,42 +121,39 @@ class Model(BaseModel):
         Starting time of the simulation.
     endtime : Union[str, datetime.datetime]
         End time of the simulation.
-    allocation : Optional[Allocation]
-        Allocation settings.
     solver : Optional[Solver]
         Solver settings.
     logging : Optional[logging]
         Logging settings.
     """
 
-    node: Node
-    edge: Edge
-    basin: Basin
-    fractional_flow: Optional[FractionalFlow]
-    level_boundary: Optional[LevelBoundary]
-    flow_boundary: Optional[FlowBoundary]
-    linear_resistance: Optional[LinearResistance]
-    manning_resistance: Optional[ManningResistance]
-    tabulated_rating_curve: Optional[TabulatedRatingCurve]
-    pump: Optional[Pump]
-    outlet: Optional[Outlet]
-    terminal: Optional[Terminal]
-    discrete_control: Optional[DiscreteControl]
-    pid_control: Optional[PidControl]
-    user: Optional[User]
     starttime: datetime.datetime
     endtime: datetime.datetime
-    allocation: Optional[Allocation]
-    solver: Optional[Solver]
-    logging: Optional[Logging]
-
-    class Config:
-        validate_assignment = True
 
-    @classmethod
-    def fields(cls):
-        """Return the names of the fields contained in the Model."""
-        return cls.__fields__.keys()
+    update_timestep: float = 86400
+    relative_dir: str = "."
+    input_dir: str = "."
+    results_dir: str = "."
+
+    database: Database = Database()
```

Comment:
```suggestion
    database: Database = Database()
```

Why is `= Database()` necessary? Isn't that the default behavior?
Or did that change with Pydantic v2?

Reply:
It was never default behaviour, but this is the way to set the default, if none is provided. If it isn't set (only var: type), it is required to be provided on init. While this is the case if we open a toml, we also need to be able to start from scratch.

Reply:
I see, this is for the interactive use case.
If I understood you correctly the following makes it optional

```python
database: Database | None = Database()
```

But what is the difference between the following?

```python
database: Database
```

and 

```python
database: Database = Database()
```

Reply:
So the `= something` provides a default, which is used if nothing is provided on init.

So `thing()` will fail in the first case with database not provided. In the second case, `thing()` will work, and the database field is set with the default. In your optional case, `thing(database=None)` would be a valid option as well.